packages/circuit-ui/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -90,21 +99,78 @@ const inlineMobileStyles = ({
     ${getInlineStyles({ theme })}
   `;
 
+const actionStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  width: 100%;
+
+  > button,
+  > a {
+    width: auto;
+
+    &:not(:last-child) {
+      margin-right: ${theme.spacings.mega};
+      margin-bottom: 0;
+    }
+  }
+
+  ${theme.mq.untilKilo} {
+    flex-direction: column-reverse;
+
+    > button,
+    > a {
+      width: 100%;
+
+      &:last-child {
+        margin-bottom: ${theme.spacings.mega};
+      }
+    }
+  }
+`;
+
 const Wrapper = styled('div')<ButtonGroupProps>(
   baseStyles,
   alignmentStyles,
   inlineMobileStyles,
 );
 
+const ActionsWrapper = styled('div')<ButtonGroupProps>(actionStyles);
+
 /**
  * Groups its Button children into a list and adds margins between.
  */
 export const ButtonGroup = forwardRef(
-  ({ children, ...props }: ButtonGroupProps, ref: ButtonGroupProps['ref']) => (
-    <Wrapper {...props} ref={ref}>
-      {children}
-    </Wrapper>
-  ),
+  (
+    { children, action, ...props }: ButtonGroupProps,
+    ref: ButtonGroupProps['ref'],
+  ) => {
+    if (action) {
+      return (
+        <ActionsWrapper {...props}>
+          <Button
```

Comment:
Accepting any children is too flexible IMO. As you noted above, the constraints enforced by the object prop enable us to better control the look of the buttons. The object prop also makes for a  better API for components that wrap the ButtonGroup such as the notification components.

Reply:
Agreed, in this case each `Action` will need to accept extra props: `isLoading`, `loadingLabel`, `destructive`, and so on (basically all LoadingButton props, since that extends the Button)

Reply:
Ah it already extends `Button`, so we probably simply need to switch this to `LoadingButton` (?) let's see if TS is okay with that

Reply:
Honestly, I think we could merge the LoadingButton into the Button component. Historically, the LoadingButton was split out because it had quite a bit of extra code, but it has been simplified significantly since then.

Would make for a better developer experience and make the types easier. 

(This would also raise the question what to do with the IconButton — could it also be folded into the Button component?)

Reply:
As discussed, let's migrate the buttons together but keep the IconButton separate for now (it has quite different requirements in terms of labeling)

Reply:
Merged the Button and LoadingButton together, and kept the LoadingButton implementation. You can check it here [8687af8](https://github.com/sumup-oss/circuit-ui/pull/1167/commits/8687af8646586d6209c3a7318590387c4530f5f6). Let me know what do you think, and if there is something else to add? :) 
I also moved some of the specs and stories of LoadingButton to Button, but also kept the ones in LoadingButton.

Reply:
And also other changes, in case I missed something 🤍